src/libraries/Microsoft.Extensions.Http/src/Logging/LoggingHttpMessageHandler.cs

```diff
@@ -102,7 +102,7 @@ public static class EventIds
 
             public static void RequestStart(ILogger logger, HttpRequestMessage request, Func<string, bool> shouldRedactHeaderValue)
             {
-                _requestStart(logger, request.Method, request.RequestUri, null);
+                _requestStart(logger, request.Method, request.RequestUri.AbsoluteUri, null);
```

Comment:
```suggestion
                if (logger.IsEnabled(LogLevel.Information))
                {
                    _requestStart(logger, request.Method, request.RequestUri.AbsoluteUri, null);
                }
```

We should check the `LogLevel` before accessing `AbsoluteUri` to avoid allocating it if logging is disabled.

Reply:
\+ in the `LoggerMessage.Define` the overload that takes `LogDefineOptions` should be used, so that `SkipEnabledCheck = true` can be set, to avoid repeated checks of `logger.IsEnabled`.

Unfortunately `LogDefineOptions` is a reference type, so that instance needs to be cached (in a static readonly, so it can be re-used in this class).

Reply:
Updated with the suggested checking. 👍